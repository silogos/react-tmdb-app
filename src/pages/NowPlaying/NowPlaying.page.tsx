import PageLayout from "@/layouts/Page";
import MovieListComponent from "@/components/MovieList";
import useMovies from "@/hooks/useMovies";

function NowPlayingPage() {
  const { data, isLoading, isError, hasMoreData, fetchNextData } = useMovies({
    path: "/movie/now_playing",
  });

  return (
    <PageLayout
      title="Now Playing Movies"
      titleAppendAppName={true}
      backgroundImagePath={data[0]?.backdrop_path}
    >
      <MovieListComponent
        movies={data}
        isLoading={isLoading}
        isError={isError}
        hasMoreData={hasMoreData}
        onLoadMore={fetchNextData}
      />
    </PageLayout>
  );
}

export default NowPlayingPage;
