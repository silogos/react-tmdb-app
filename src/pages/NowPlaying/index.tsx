export { default } from "./NowPlaying.page";
