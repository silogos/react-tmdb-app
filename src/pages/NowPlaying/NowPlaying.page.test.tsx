import { describe, it, expect } from "vitest";
import NowPlaying from "./NowPlaying.page";
import { render } from "@testing-library/react";

describe("NowPlaying Page", () => {
  it("renders correctly", () => {
    const component = render(<NowPlaying />);
    expect(component).toBeTruthy();
  });

  it("renders toMatchSnapshot", () => {
    const component = render(<NowPlaying />);
    expect(component).toMatchSnapshot();
  });
});
