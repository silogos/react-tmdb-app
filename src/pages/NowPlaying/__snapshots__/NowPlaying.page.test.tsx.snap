// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NowPlaying Page > renders correctly 1`] = `
<div>
  <div
    data-testid="MovieList"
  />
</div>
`;
